raster-foundry/raster-foundry
# Add batch job to remove scenes from database

## Overview

*conflicts with #5503 for migration numbers*

This PR adds a batch job to remove scenes in a datasource from a date to the beginning of the next month.

### Checklist

- [x] ~New~ old tables and queries have appropriate indices added
- ~Any new SQL strings have tests~ (testing instructions for a one-off job)

### Demo

```bash
$ ./scripts/console batch "java -cp /opt/raster-foundry/jars/batch-assembly.jar com.rasterfoundry.batch.Main remove-scenes 697a0b91-b7a8-446e-842c-97cda155554d 2017-03-01"

Creating raster-foundry_batch_run ... done
[ioapp-compute-8] INFO  c.r.b.r.RemoveScenes - Removed 1814 scenes for datasource 697a0b91-b7a8-446e-842c-97cda155554d
```

### Notes

The foreign key indices on images, thumbnails, and bands are going to take a while to create, so I'm creating them concurrently. They shouldn't run into much in terms of conflicts with the scenes table with the lambdas for public data import disabled, but they're still going to create some pretty big indices. Unfortunately they're necessary, since the difference between deleting with the indices and without even for the small size of the local db is significant -- with the indices the job took 13 seconds to run for March 2017 Landsat 8, and without it it was on about 3 minutes before I killed it. >= 12x speedup for 60 months of sentinel-2 and 84 months of Landsat 8 will be worth it.

I'm gonna hold off on changelog until after #5503 because `chan`

## Testing Instructions

- count Landsat 8 scenes in your database: `select count(1) from scenes where datasource = '697a0b91-b7a8-446e-842c-97cda155554d'`
- assemble batch jar
- bring up postgres and memcached -- `docker-compose up -d postgres memcached`
- run the command in the demo section
- count Landsat 8 scenes in your database again -- the number remaining should be your initial count - how many the job reports it deleted

Closes azavea/raster-foundry-platform#1104
